docs: qtvcp-* reformatting

silopolis opened the pull request:
* Fix missing blank lines between includes in Master_Developer.adoc (not required but good practice as AsciiDoc doesn't do it when files are actually included)
* Fix qtvcp_*.adoc formating broken while translation migration

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -141,110 +175,164 @@ Such libraries include:
-change the basename in the edit box.
+If you wish to *modify a stock screen* with full control, copy it's UI
+and handler file to your configuration folder. There is a QtVCP panel to
+help with this.
```
Here you left out what was before in line 212 since this was too simple?

smoe commented:
I think you did a very nice review, @silopolis! I already want to thank you for this and am confident that @c-morley also likes it or has more ideas such that he would also like it.

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -2,72 +2,93 @@
+There are two files that can be used, individually or in combination to
+add customization:
+
+* A *UI file* that is made with QT's Designer graphical editor.
```
Qt Designer

silopolis replied:
Fixed thanks

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -2,72 +2,93 @@
+requirements.
+
+One is not restricted to adding a custom panel on the right or a custom
+tab as QtVCP leverages _QT Designer_ (the editor) and _PyQT5_ (the
```
Qt Designer

silopolis replied:
Fixed thanks

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -2,72 +2,93 @@
 
-Qtvcp uses the PyQt5 toolkit's widgets for linuxcnc integration.
-Widget is the general name for objects such as buttons and labels in PyQT5.
 You are free to use any available widgets in the QTDesigner editor.
```
Qt Designer

silopolis replied:
done

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -101,34 +124,45 @@ GRAPHICS_CYCLE_TIME = 100
+
+PyQt5 uses this file to build the display and react to those widgets.
+
+The QTDesigner editor makes it relatively easy to build and edit this
```
Qt Designer

silopolis replied:
done

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-Then connect any required HAL pins in a HAL file
+To build a panel or screen:
+
+* Use QTDesigner to build a design you like and save this design to your
```
Qt Designer

silopolis replied:
done

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
 
-You must have designer installed; These commands should add it:
-Or use your package manager to install the same:
+First you must have designer installed. The following commands should
```
the Qt Designer (executable `designer`)

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-You may need to add the plugins/python folders
+You may need to create the `plugins/python` folder.
 
 To start Designer:
```
Qt Designer

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-then load designer with : 'designer -qt=5'
+* For a _RIP install_: +
+  Open a terminal, set the environment for LinuxCNC <1>, then load
+  designer <2> with :
```
I am not ultimately confident that I would want to read "Qt" here again, maybe better as "then load the designer..."?

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
+Now the button will turn the machine on when pressed.
 
-And now the direct way with Designer's property editor
+And now the direct way with Designer's property editor:
```
Qt Designer - confident.

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-  You will see a list of properties and values. find the 'machine on action' and
-  click the checkbox. the button will now control machine on/off.
+* Select the 'Machine On' button.
+* Go to the Property Editor on the right side of Designer.
```
the Qt Designer

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -758,82 +937,93 @@ designer editor.
+Designer.
 
 [[cha:designer-slots]]
 === Using Designer to add slots
```
Qt Designer

silopolis commented:
> Sorry, I should have stopped a bit earlier

Nop mate, was fun to try to fix them faster than you were spotting these ;-)

> I am not sure about how much effort you should invest upfront, let us just hope that @c-morley loves this much as we both do.

I hope too !
Just did qtvcp.adoc as a proposal. Will do the rest once I get style agreement...

c-morley commented:
Ya guys - of course you have done a wonderful job. Very nicely done.

hansu commented:
Is it ready to merge?

smoe commented:
Yes. Ready to merge, I am confident. I am somewhat uncertain about how this will effect the .po files but whatever we do needs to be done manually and within weblate it is likely easier than to go through the .po file generation again.

silopolis commented:
Le mar. 17 mai 2022 à 13:28, Steffen Möller ***@***.***> a
écrit :

> Yes. Ready to merge, I am confident. I am somewhat uncertain about how
> this will effect the .po files but whatever we do needs to be done manually
> and within weblate it is likely easier than to go through the .po file
> generation again.
>
Wowowow, really ?!

First I thought this was for after the merger as this completely breaks
sync on these files and I thought we'd better stay in sync until everything
is rolling smoothly up to published, auto generated, translated docs.
Then, this is only the first chapter of the massive QtVCP documentation.
Reached 1/4th of qtvcp-widgets.adoc in the meantime...

smoe commented:
@hansu  was asking because of weblate, I think. You got the green light from @c-morley to continue, which was the important bit. I am very happy to have the QtVCP documentation as a brainstorming branch for the style guide. But we will not finish this prior to weblate - I hope :)

Since you have done so much good already, I would not like to see this missing from the upcoming merge to master, this is why I said "ready". I suggest you push what you have locally and then branch off from master with another PR to continue.

silopolis commented:
Le mar. 17 mai 2022 à 13:49, Steffen Möller ***@***.***> a
écrit :

> @hansu <https://github.com/hansu> was asking because of weblate, I think.
> You got the green light from @c-morley <https://github.com/c-morley> to
> continue, which was the important bit. I am very happy to have the QtVCP
> documentation as a brainstorming branch for the style guide. But we will
> not finish this prior to weblate - I hope :)
>
I hope too 😅

> Since you have done so much good already, I would not like to see this
> missing from the upcoming merge to master, this is why I said "ready". I
> suggest you push what you have locally and then branch off from master with
> another PR to continue.
>
I've only done a thousand lines and still 6000 left to reformat the whole
QtVCP documentation, so don't hold your breath !
That widget chapter is 2400 lines alone...

Put aside that I prefer to have a single PR, if not commit, for the
complete  "upgrade", that could be a good candidate for our first
translation tool chain crash test !

hansu commented:
Ok just push everything you have locally, then remove the draft status to give the go for merging.
But I think I won't manage to do something on merging docs-devel into master before the weekend, so you still have some time

hansu commented:
So I'll merge this now?

smoe commented:
@silopolis  - And if you just go for the html instead of css? Maybe followed by a FIXME: comment that this should be css for real? Then we can easily find this whenever we have in place whatever package/link is missing.

silopolis commented:
@smoe my question would rather be: why doesn't it finds the language where as it actually is in the source-highlight/local/ directory ?!

smoe commented:
> @smoe my question would rather be: why doesn't it finds the language where as it actually is in the source-highlight/local/ directory ?!

So this builds for you, you are saying? Since the change is trivial and purely technical, this can be done easily for all languages. There is no point in delaying the merge over this, so I would just go and get to something that works and do anything more advanced later.

smoe commented:
Builds again, yippee. These are 81 minutes build time are ... bad ... but good enough, let's merge. Everything else can happen later, especially the improvement of optics. My concern is that the community is aware of much happening on the documentation and consequentially does not work on improvements that the community would otherwise address.

hansu commented:
What's the status of this? Perhaps it would be good to merge it at some point if you don't have time to continue with this in the near future to avoid double work or conflicts when merging.

smoe commented:
@silopolis , I agree. What is better than what is already available should be merged early.

silopolis commented:
Le dim. 19 juin 2022 à 14:43, Steffen Möller ***@***.***> a
écrit :

> @silopolis <https://github.com/silopolis> , I agree. What is better than
> what is already available should be merged early.
>
Very true! Will finish this this week...

petterreinholdtsen commented:
This is a horribly long patch that is hard to make heads and tails of.  For example, why are so many manual pages deleted?  Could it be split up into smaller chunks?

hansu commented:
> For example, why are so many manual pages deleted? Could it be split up into smaller chunks?

I think this is because this branch is based on a commit before the docs-devel merge. After a rebase to current master it would look much cleaner I guess.

silopolis commented:
Le ven. 15 juil. 2022 à 11:52, Hans ***@***.***> a écrit :

> For example, why are so many manual pages deleted? Could it be split up
> into smaller chunks?
>
> I think this is because this branch is based on a commit before the
> docs-devel merge. After a rebase to current master it would look much
> cleaner I guess.
>

Most probably. This PR only really holds edits to QtVCP's adocs files and a
handful of others. I never touched any man pages in any of my works, I
believe.
Also, I not sure this is up to date with the latest mods pushed.
Sadly, I'm AFK until at least the end of tomorrow.
Will update this in one way or another by the end of the WE...

hansu commented:
@silopolis what are your plans to continue this?

silopolis commented:
Le dim. 28 août 2022 à 15:58, Hans Unzner ***@***.***> a
écrit :

> @silopolis <https://github.com/silopolis> what are your plans to continue
> this?
>

Yes!
Just, like I said to Petter in the channel this week, I hope I can tackle
this in the comming week. Hope the ton of commits merged since won't make
this a galore...

petterreinholdtsen commented:
[Jérémie Tarot]
> Hope the ton of commits merged since won't make
> this a galore...

I suspect the easiest way forward is to create a new branch from master
and cherry-pick the relevant commits you want to carry forward.
Rebasing and merging only gave me problems.

-- 
Happy hacking
Petter Reinholdtsen

silopolis commented:
Le dim. 28 août 2022 à 19:31, petterreinholdtsen ***@***.***>
a écrit :

> [Jérémie Tarot]
> > Hope the ton of commits merged since won't make
> > this a galore...
>
> I suspect the easiest way forward is to create a new branch from master
> and cherry-pick the relevant commits you want to carry forward.
> Rebasing and merging only gave me problems.



Indeed looks like the best strategy! 👍
Thanks master 🙏😉

silopolis commented:
I've finally finished and fixed most of my WS migration issues and found
enough time to get back to our matters, yey !

I have completed yesterday the rebase and fix/update of qtvcp.adoc and
qtvcp-widgets.adoc massive reformatting.
Now I need your opinion, and possibly (git) help to go forward...
If I'd like to push in this PR to keep all the mods to QtVCP docs in one PR
and attached to this long running discussion, this raises two issues:
1. Modifications a so important that I fear to turn this into a review
nightmare. Should we just drop this PR and go with a per file commit
strategy ?
2. I followed Petter's advice to start fresh on a new master branch and
cherry pick my commits, but I'm unable to reset my local branch to master.
Is it something possible ? Like forcing a rebase to discard all conflicts ?
    Or can I switch the source branch for this PR ? I thought I had already
done this but can't find back how ! :/

Thanks for help and patience guys 🙏

petterreinholdtsen commented:
[Jérémie Tarot]
> 2. I followed Petter's advice to start fresh on a new master branch and
> cherry pick my commits, but I'm unable to reset my local branch to master.
> Is it something possible ? Like forcing a rebase to discard all conflicts ?
>     Or can I switch the source branch for this PR ? I thought I had already
> done this but can't find back how ! :/

Not quite sure I understand the question, but will try to answer anyway.
I assume you got a local branch docs/formating_fixes which is pushed to
github as the basis of this pull request, and a new local branch X based
on a recent master which  you want to become the basis for this pull
request.  To do this, you need to rename the branch names locally and
force push the new branch to github with the old name.  Ie something
like this:

  git branch -m docs/formating_fixes docs/formating_fixes-orig
  git branch -m X docs/formating_fixes
  git push githubremote docs/formating_fixes -f

I do not know what your git remote name for your github fork is, but
assume it is githubremote above.
-- 
Happy hacking
Petter Reinholdtsen

hansu commented:
Or just push your new local branch to the old remote branch like
`git push -u githubremote formating_fixes_new:githubremote/formating_fixes --force`

Where githubremote is the name of your remote. origin by default

silopolis commented:
Le jeu. 8 sept. 2022 à 11:30, petterreinholdtsen ***@***.***>
a écrit :

> I suspect the new branch is not quite right, as it got more than a
> thousand commits, including commits originating from Weblate. I notice it
> is relative to docs-devel, not master.
>
Fixed !
Thank you, and thank you @hansu, used both of your tips to switch branches
👍🙏

silopolis commented:
Any idea why PDFs build chokes on CSS syntax highlighting ?

hansu commented:
> Any idea why PDFs build chokes on CSS syntax highlighting ?

CSS is not supported by the listings package.
See https://ftp.acc.umu.se/mirror/CTAN/macros/latex/contrib/listings/listings.pdf page 13

hansu commented:
But you can add `:css: {basebackend@docbook:'':css}` at the top and then use `[source,{css}]` analog to the custom defined languages.

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -2,100 +2,133 @@
 image::images/qtaxis.png["QTscreen Qtaxis",align="left"]
-.Blender - 4 Axes Sample
+
+.Blender - 4 Axis Sample
```
"4 Axes" was correct as 'axes' is the plural of 'axis'.
Was a bit inconsistent, before.

silopolis replied:
​Absolutely ! My bad. That error is seen so often that tends to stick at times :/​

hansu commented:
Otherwise I know not much about QtVCP to evaluate this.

hansu commented:
> Le jeu. 8 sept. 2022 à 16:41, Hans Unzner ***@***.***> a écrit :
> But you can add :css: ***@***.***:'':css} at the top and then use [source,{css}] analog to the custom defined languages.
> Nice, but where would it take definitions ? In the syntax-highlighting directory ?

It would just leave the PDF without syntax highlighting as there is nothing suitable available.

silopolis commented:
Le jeu. 8 sept. 2022 à 19:07, Hans Unzner ***@***.***> a
écrit :

> Le jeu. 8 sept. 2022 à 16:41, Hans Unzner *@*.
> *> a écrit : But you can add :css: @.*:'':css} at the top and then use
> [source,{css}] analog to the custom defined languages.
> Nice, but where would it take definitions ? In the syntax-highlighting
> directory ?
>
> It would just leave the PDF without syntax highlighting as there is
> nothing suitable available.
>

Where are our custom hal etc definitions then?

hansu commented:
> Le jeu. 8 sept. 2022 à 19:07, Hans Unzner ***@***.***> a écrit :
> Le jeu. 8 sept. 2022 à 16:41, Hans Unzner *@*. *> a écrit : But you can add :css: @.*:'':css} at the top and then use [source,{css}] analog to the custom defined languages. Nice, but where would it take definitions ? In the syntax-highlighting directory ? It would just leave the PDF without syntax highlighting as there is nothing suitable available.
> Where are our custom hal etc definitions then?

`docs/src/source-highlight/{hal|ini|ngc}.lang`
But these are only for the HTML output. I am not sure how `docs/src/source-highlight/emc-langs-source-highlight.conf` is involved here.
For PDF, I can only refer to this comment:
```
// for now, PDF's can't have highlighted ini,hal or ngc files
// for brave souls: extend /usr/share/texmf-texlive/tex/latex/listings/lstlang1.sty
// and make it a local copy in the current directory
// HTML works fine
```

smoe reviewed (commented):
Well done. We have the axis <-> axes quirk  still pending but can fix that later.

smoe commented:
The build errors seem trivial to fix. So, many thanks!

andypugh commented:
Whilst I agree that axes is the plural of axis (and, confusingly, of axe) the conventional usage is 3-axis, 4-axis etc and I think that trying to correct this point of grammar is likely to cause more confusion and less effective searching. 
It isn't that unusual, native speakers will talk of a "three car garage" and a "five kilometer walk". I can't really explain why this is, though possibly it is because the item being described is singular. 
https://books.google.com/ngrams/graph?content=five+mile+walk%2C+five+miles+walk&year_start=1800&year_end=2019&corpus=en-2019&smoothing=3
Shows this fairly clearly.

hansu commented:
Sorry for pointing you into a wrong direction with the "axes". Good that we have some native speakers here :-).
According to google, writing it with a hyphen like "4-Axis" seems the most common form.

silopolis commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -13,38 +13,42 @@
 
-.X1mill - 4 Axis Sample
-image::images/x1mill.png["QTscreen x1mill",align="left"]
+.X1mill - 4 Axes Sample
```
me too ! :/

silopolis commented:
Le ven. 9 sept. 2022 à 14:10, Hans Unzner ***@***.***> a
écrit :

> Sorry for pointing you into a wrong direction with the "axes". Good that
> we have some native speakers here :-).
> According to google, writing it with a hyphen like "4-Axis" seems the most
> common form.
>
No problem :)

So, I finally reached the point where I've done my best at reformatting the
qtvcp-* files...

I've stripped out a couple of non relevant files from this PR (thank you
guys for merging these right away) and squashed as much as felt appropriate.

This remains massive so forgive me for the reviewing chore... Hope this
will fit your taste even if I may have fallen in some kind of over inline
formatting trying to emphasize the important and significant parts of the
text.

A few ideas and beliefs come out of this work:
* Those files are _way_ too big and should be split. This is the case for
many of our chapters !
* Nesting such an amount of content into the global docs tree makes it
difficult to structure optimally due to the limitation to an overall 5
levels ToC tree.
  Being an application by itself, for the best of all, I believe QtVCP
(like the other apps) should have their own manual/guide. The main docs
should just have a chapter listing those apps with a small summary and a
link to their own docs.
  Note that as far as QtVCP is concerned, I believe that it would benefit
(like QtPyVCP) from having its own repository to free itself from LinuxCNC
release cycle etc. But that's another discussion.
* There is duplicate and/or split content that should be reorganized (see
Vismach related sections for example)
* last but not least, a more than significant part of this is really code
documentation that should be directly generated from code embedded
documentation using tools like sphinx and doxygen.

Feverishly waiting for your feedback, especially you @cmorley for this
part, this has triggered another bunch of work on other files that I'd like
to push out of my local stashes...

Have a beautiful we my friends

TY
J

c-morley commented:
It looks good to me -what a lot of work!
-split big files - How would you split the file more, can you give an example? In fact I split qtvcp into many parts for this very reason - compare it what we did with gladvcp.
- I noticed TOC didn't really differentiate  when getting too deep - not sure if that could be improved.
- Not sure how a separate manual really helps things - could you explain why?
- Qtvcp is really tied to the linuxcnc release cycle so a separate repository just adds overhead, for little up side (I had considered this myself at one time) QtPyVCP is a separate project all together, hence separate repo.
- split/duplicate content - This is difficult. take vismach - these are two different code bases with small but significant differences. They do obviously come from the same base code. I predict they will diverge in code the more the qt version is used (as it's embedable). I think it's messy to put the docs together and split out the differences with NOTEs.
- I really know nothing of code/doc generators - though I bet it would force us to doc the code directly more - which would be a bonus I suppose. I wonder if that would make working in the code more painful.

silopolis commented:
Le sam. 10 sept. 2022 à 20:46, c-morley ***@***.***> a écrit :

> It looks good to me - what a lot of work !
>

Very glad you like it !
Was indeed a lot of work but was a very interesting read and hope it will
benefit the project...

-split big files - How would you split the file more, can you give an
> example? In fact I split qtvcp into many parts for this very reason
>

I'd start with splitting sections into independent files and create a qtvcp
dedicated directory.

compare it what we did with gladvcp.
>

This is indeed one of those 3K+ lines files that must be split too !


>    - I noticed TOC didn't really differentiate when getting too deep -
>    not sure if that could be improved.
>
>
Current documentation structure doesn't allow to go deeper than three
levels in individual chapters. For reference content, this forbids more
detailed ToCs giving more direct access to specific content.


>    - Not sure how a separate manual really helps things - could you
>    explain why?
>
> By giving more direct access to specific GUIs documentation, creating
individual and identifiable guides for each project component and avoid a
frightening 1K+ pages printable manual !


>    - Qtvcp is really tied to the linuxcnc release cycle so a separate
>    repository just adds overhead, for little up side (I had considered this
>    myself at one time) QtPyVCP is a separate project all together, hence
>    separate repo.
>
>
Hmm, on second thought, I may have confused QtVCP based apps release cycle
with the framework's one.


>    - split/duplicate content - This is difficult. take vismach - these
>    are two different code bases with small but significant differences.
>
>
I just meant there was split content I QtVCP's docs that _maybe_ could be
regrouped ?


>    - I really know nothing of code/doc generators - though I bet it would
>    force us to doc the code directly more - which would be a bonus I suppose.
>    I wonder if that would make working in the code more painful.
>
>
Just one of the zillion references available
https://realpython.com/documenting-python-code/
TY
J

petterreinholdtsen commented:
I do not know AsciiDoc formatting well enough to judge the content of this change, but believe we should either trust @silopolis on this or someone should volunteer to review the changes quickly to find any problems and get them fixed in a reasonable time frame.

hansu commented on docs/src/gui/qtvcp-vcp-panels.adoc:
```diff
@@ -244,22 +257,9 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-<<cha:qtvcp-development,QtVCP Development>>
-
-<<cha:qtvcp-custom-widgets,QtVCP Custom Designer Widgets>>
+<1> In this case we load `qtvcp` using *`-Wn`* which waits for the panel
```
Is `<1>` needed in the code? 
And I guess "1."  should be a footnote referring to "<1>" ?
If I am right, it is more guessed and not really clear.

silopolis replied:
See my other answer about callouts

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -267,197 +387,276 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-In this case we load qtvcp using -Wn; which waits for the panel to finish loading before
-continuing to run the next HAL command. This is so the HAL pins from the panel are finished
-in case the are used in the rest of the file.
+<1> In this case we load `qtvcp` using *`-Wn`* which waits for the panel
```
Is <1> needed in the code?
And I guess "<1>" should be a footnote referring to "<1>" ?
Unfortunately "<1>" is formatted as "1."

![grafik](https://user-images.githubusercontent.com/67957916/189978014-c2ea11c6-c4ba-4d0a-bf8d-0b93082e3b60.png)

So not really clear.

silopolis replied:
Yes "<1>" must be in the code block. See https://docs.asciidoctor.org/asciidoc/latest/verbatim/callouts/

About formatting, maybe this can be customized to our likings, but this is quite common rendering.

hansu replied:
I wouldn't associate the <1> with the list item "1."
In the linked examples it is formatted both as  ① which makes it clear.
And they added the <1> as a comment in the code in the second example.
So if this isn't supported by asciidoc-py, we should format it in another way to make it clear.
First have equal symbols and second maybe add the explanation as a footnote.

silopolis replied:
I fixed the missing comment character.

From https://asciidoc-py.github.io/userguide.html#X105 it seems we're missing the `-a icons` command-line option for fine rendering.

hansu replied:
>From https://asciidoc-py.github.io/userguide.html#X105 it seems we're missing the -a icons command-line option for fine rendering.

Great! That seems to resolve this issue.

silopolis replied:
I just need to be pointed to the right place if we want this option to be added with this PR or a later (dedicated ?) one... Please tell me.

hansu replied:
I'll check that and commit it separately.

hansu replied:
Looks like a bigger change. `-a icons` tells asciidoc to use icons globally, so for icons like [NOTE], [EXAMPLE] etc. as well.
And as we don't have them, it complains about it. And we have to add icons for (1), (2) etc. or adapt the preprocessor to use the unicode characters for that.

hansu replied:
Yeah, but than we have to copy them somewhere and make sure that they are on the right place for all languages and so on.

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -267,197 +387,276 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
 ////
 
-Then you must add a link to the `qtvcp_plugin.py` to the folder that designer will search.
+.Add `qtvcp_plugin.py` link Qt Designer Search Path
```
I guess so ?
```suggestion
.Add `qtvcp_plugin.py` link to Qt Designer Search Path
```

smoe replied:
to -> to the

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -669,176 +868,208 @@ def get_handlers(halcomp,widgets,paths):
+NOTE: It is usually better to use the special `closing_cleanup__`
+  function.
+
+== Connecting Widgets To Python Code
```
```suggestion
== Connecting Widgets to Python Code
```

smoe replied:
+1

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -669,176 +868,208 @@ def get_handlers(halcomp,widgets,paths):
+* You _create the slot's functions_ defined in Qt Designer.
+
+[[sub:qtvcp:designer-slots]]
+=== Using Qt Designer To Add Slots
```
```suggestion
=== Using Qt Designer to Add Slots
```
or
```suggestion
=== Using Qt Designer to add Slots
```
? 
I am not sure.

smoe replied:
capitalized (even though I dislike this for level three titles)

silopolis replied:
As said in previous comment, reading the excellent GitLab style guide and referenced documents makes me believe we should get away from camel capitalized titles and settle on sentence capitalization.
What remains less clear to me is the capitalization of "special" words, like "Slots" in this case...

silopolis replied:
Reading again https://docs.gitlab.com/ee/development/documentation/styleguide/#feature-names I believe "Slots" should indeed be title cased.

hansu commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -669,176 +868,208 @@ def get_handlers(halcomp,widgets,paths):
+.Qt Designer Signal/Slot Selection
+image::images/designer_slots.png["Qt Designer Signal/Slot Selection",align="center"]
+
+=== Handler File Changes
```
"Handler File Changes"

Maybe here can be found a better heading? This heading is more confusing than helpful IMHO.

smoe replied:
Handler file changes are those changes that are performed by the handler to files. File Handler changes are changes performed to the file handler. Just keep it like the original and accept, I suggest.

hansu replied:
Yeah just want top improve the understanding of the original...

hansu commented on docs/src/gui/qtvcp-widgets.adoc:
```diff
@@ -75,543 +96,676 @@ LED #name_0f_led{
 
-=== Action Button Widget
+[[sub:qtvcp:widgets:actionbutton]]
+=== `ActionButton` Controller Action Control Widget
```
In my opinion all the headings are too long. This makes the table of contents appearing more as block of text than a list for getting an overview.
I would keep the headings short and add the correct name and the description inside the appropriate section.

![grafik](https://user-images.githubusercontent.com/67957916/189980583-2274f3d2-1e0f-403d-a445-42833158d6ae.png)

silopolis replied:
As the desciptive part of these titles is already in first line of all these sections, those could be shortened to the bare widget name.
The new reader may in turn have to dig a little deeper to grasp the purpose of each widget... but the Fine Manual thou shall read ! ;)

smoe replied:
A separator like a dash may be helpful. I just find them borerlinish-long, not to long. But yes, this is beyond "optics only".

silopolis replied:
The added dash indeed helps a bit for readability.

hansu replied:
Just a stupid question - what should the part after the dash say to me?
Like "Machine Controller Action Control Widget"...
Is it something that has to be used in the code when using these widgets?

hansu replied:
If it's a description then one more reason to shift it inside the section and away from the title IMHO.
But it's readable now, so we can keep it.

silopolis replied:
And like I said, this small description is already on the first line of each section...

@c-morley what about stripping out the small desc from titles ? Are you OK with that ?

hansu commented on docs/src/gui/qtvcp-vismach.adoc:
```diff
@@ -1,48 +1,57 @@
-Vismach displays the model in a 3D viewport and the model
-parts are animated +
-as the values of associated HAL pins change.
+*Vismach* is a set of *Python functions that can be used to create and
```
This text could have a section heading as well like you did in qtvcp.adoc.

silopolis replied:
You mean `[[...]]` anchors ?

silopolis replied:
Assumed yes and added title anchors.

hansu replied:
No, a heading as I said.
```suggestion
== Overview / Intro...
*Vismach* is a set of Python functions that can be used to create and
```

silopolis replied:
OK, done.

hansu commented on docs/src/gui/qtvcp-vismach.adoc:
```diff
@@ -127,123 +138,159 @@ c.ready()
 ----
 
-== Orientating Model Parts
+== Moving Model Parts
```
Maybe "Transforming"?
Is a rotation also a movement? 
Maybe ignore this comment.

silopolis replied:
I believe rotation is indeed also a movement, but may be that "Transformation" is a more mathematicaly accurate word ?
OTOH, some may interpret "transformation" as a change in parts shape instead of position, but that can be explicited...
What do others think ?

silopolis replied:
As this is wording related and as a full content centric pass could be done. I leave it for the next one.

hansu commented on docs/src/gui/qtvcp-vismach.adoc:
```diff
@@ -127,123 +138,159 @@ c.ready()
+* as it is moved by the Vismach script during model assembly, or
+* as it moves in response to the HAL pins as the model is animated.
+
+=== `HalTranslate`
```
In general I think we shouldn't have a mono-space font in numbered headings. But it's more a personal opinion.

silopolis replied:
Like I said in another message, mono-space font looks adequate to me to style all commands, paths, files, functions, keywords... But the styles for titles should be improved to make it smoother to the eyes, like having the same styling as the rest of the title to start with

hansu commented on docs/src/gui/qtvcp-vismach.adoc:
```diff
@@ -298,51 +346,54 @@ For a _parallel machine_ there may be several "base" parts.
+    It is advisable to do as the default initial viewpoint is rather
+    unhelpfully from immediately overhead.
+
+== Basic structure of a QtVismach script
```
```suggestion
== Basic Structure of a QtVismach script
```

silopolis replied:
Titles capitalization is really something we have to decide !

With my afternoon readings, I believe we should follow simple "Sentence" capitalization and only raise first character.

smoe replied:
If *S*tructures then also *S*cript.

We can fix this any time, but yes, we need to decide about it. Would leave that decision to the English natives, though.

silopolis replied:
I'll follow https://docs.gitlab.com/ee/development/documentation/styleguide/#markdown-rule-md044proper-names-capitalization and https://docs.gitlab.com/ee/development/documentation/styleguide/#capitalization as they looks perfectly good to me.

hansu commented on docs/src/gui/qtvcp-vcp-panels.adoc:
```diff
@@ -244,22 +257,9 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
-
-
-== More Information
-
-<<cha:qtvcp-widgets,QtVCP Widgets>>
-
-<<cha:qtvcp-libraries,QtVCP Libraries>>
-
-<<cha:qtvcp-code,QtVCP Handler File Code Snippets>>
-
-<<cha:qtvcp-development,QtVCP Development>>
-
-<<cha:qtvcp-custom-widgets,QtVCP Custom Designer Widgets>>
```
Why not keeping these references?

smoe replied:
I admit to have wondered, too.

silopolis replied:
Because this is already present at the bottom of the qtvcp.adoc file. If this makes sense for the latter as a "QtVCP Guide" ToC, I don't see the point of putting it here again. Else we'd have to put it at the end of each chapter which doesn't really make sense and which we don't do anywhere else.

smoe replied:
Reduced redundancies are fine with me.

hansu replied:
I saw it more as a cross reference as we have it in several man pages like @petterreinholdtsen was recently improving in e.g. https://github.com/LinuxCNC/linuxcnc/pull/2016.
But it's up to how you would like to structure it, just wanted to hear your thoughts 😉

silopolis replied:
I get the cross reference point and support improvements such as @petterreinholdtsen 's, but I feel they happen at a more appropriate finer grain level. Cross-referencing is a field of improvement but IMHO should be done mostly inline and at the most accurate level possible.

If really desirable we could replace this with a cross-ref to the QtVCP guide ToC... But wouldn't it be better provided with automatic Back/Up/Home/Next links ?

hansu replied:
Or add something like "for an overview of QtVCP related stuff see ...../QtVCP.html".
But this should be clear so don't have to be mentioned here.

hansu commented on docs/src/gui/qtvcp-vcp-panels.adoc:
```diff
@@ -10,192 +10,198 @@
 
-=== vismach_mill_xyz
+[[sub:qtvcp:panels:vismach]]
+=== `vismach_*` Panels
```
I would shift this on level up and have "real subsections" inside
```suggestion
== `vismach_*` Panels
```

silopolis replied:
Idea crossed my mind actually, but feared of over-structuring as this is really just a list of demo/sim panels...

But if you think so too, then it may not be that bad ;)

hansu commented on docs/src/gui/qtvcp-vcp-panels.adoc:
```diff
@@ -10,192 +10,198 @@
 
-.QtVCP vismach- 3 Axis Mill panel
-image::images/qtvismach.png["QtVCP Vismach Mill",align="left"]
+.QtVCP `vismach_mill_xyz` - 3-Axis Mill 3D View Panel
```
```suggestion
=== QtVCP `vismach_mill_xyz` - 3-Axis Mill 3D View Panel
```

and so on...

silopolis replied:
Go for it, will just keep the titles short.

hansu reviewed (commented):
On the next files I will have a look in the next days...

hansu commented:
The next files will follow...

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -2,100 +2,138 @@
+launched for the files, then in the system skin folder holding standard
+screens.
+
+.Cycles Rates
```
@silopolis only did changes to the optics, but yes, I agree - as with feeds, there is a bit of a confusion throughout about what rates and what speeds are.

hansu replied:
No, this heading is new. And wouldn't be "Cycle Times" be the correct plural instead of "Cycles Times"?

hansu replied:
I mean you introduced this heading ;-)

One cycle time --> several cycle times, see https://www.myenglishpages.com/english/grammar-lesson-plural-compound-nouns.php

silopolis replied:
okkkkk ':D
Thanks prof !

smoe commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -145,111 +195,181 @@ Such libraries include:
+The _Windows theme_ is default for screens. +
+The _System theme_ is default for panels.
+
+To see available themes load with:
```
We should not discuss content in this thread, and if it is only because I am long waiting for this to be accepted so the translations can go forward. We just do not have anywhere to discuss documentation without bothering too many with such details.

silopolis replied:
Hopefuly fixed.

@smoe agreed

silopolis commented on docs/src/gui/qtvcp.adoc:
```diff
@@ -262,195 +350,256 @@ net s32-in1        test_panel.doublescale_1-s   classicladder.0.s32in-00
+Action buttons _default to no action_ so we must change the properties
+for defined functions. +
+You can edit the properties directly in the property editor on the right
+side of designer. +
```
done

hansu commented on docs/src/gui/qtvcp-widgets.adoc:
```diff
@@ -1322,433 +1505,575 @@ self.w.screen_options.setProperty('close_overlay_color', color)
 ----
 
-It is based on the LED widget
+<1> `name_of_led` would be the name defined Qt Designer's editor.
```
...

hansu commented on docs/src/gui/qtvcp-development.adoc:
```diff
@@ -1,161 +1,206 @@
-By using the same toolkit across many screens/panels users should have an easier
-time customizing/creating and developers should find it easier to help
-trouble shoot with less effort.
+The intention of QtVCP is to *supply an infrastructure to support
+screen and VCP panel building for LinuxCNC*.
+
+By providing a _diverse widget_ set and supporting _custom coding_,
+QtVCP hopes that development energy will be expended in _one toolkit_
+rather than continuous re-invention.
+
+By using the same toolkit across many screens/panels users should have
+an easier time customizing/creating these, and developers should find it
+easier to help trouble shoot with less effort.
```
Maybe shift this paragraph into the section "Overview" ?

hansu commented on docs/src/gui/qtvcp-development.adoc:
```diff
@@ -1,161 +1,206 @@
+* the _handler file_, and
+* possibly the `.qss` _theme file_.
+
+== QtVCP Startup To Shutdown
```
```suggestion
== QtVCP Startup to Shutdown
```

hansu commented on docs/src/gui/qtvcp-development.adoc:
```diff
@@ -1,161 +1,206 @@
-mode so you can send them to ACTION functions without conversion.
+=== Error Code Collecting
+
+*LinuxCNC's error code collecting can only be read from one place*.
```
Why bold? A bit too much bold in this paragraph to highlight the parts that really should be bold.

hansu commented on docs/src/gui/qtvcp-development.adoc:
```diff
@@ -1,161 +1,206 @@
+
+=== Jog Rate
+
+*LinuxCNC has no internal record of jog rate*: *you must specify it at
```
Same as above.

hansu commented on docs/src/gui/qtvcp-custom-widgets.adoc:
```diff
@@ -87,75 +114,72 @@ Lets look at a snippet of simple_widgets.py.
+    and `_HalSensitiveBase`. +
+    Pretty standard other than the widget name will change.
+
+== Custom Controller Widgets Using `STATUS`
```
```suggestion
== Custom Controller Widgets using `STATUS`
```

silopolis replied:
...

hansu commented on docs/src/gui/qtvcp-code-snippets.adoc:
```diff
@@ -1,18 +1,22 @@
+[[cha:qtvcp:code]]
+= QtVCP Handler File Code Snippets
 
 Here are bits of ideas to put in the handler file.
```
This looks a bit lost. Can this be removed?

smoe replied:
I like a consistent formatting of achors, like [document:header] across all documents. And the start with a capital letter in the top-level heading seems also ok?

hansu replied:
This comment should be linked to line 7, which isn't obvious as I just see :D

It's about this line appearing before any section:
"Here are bits of ideas to put in the handler file."

hansu commented on docs/src/gui/qtvcp-code-snippets.adoc:
```diff
@@ -22,7 +26,8 @@ if self.w.PREFS_:
 
-Under the 'def closing_cleanup__(self):' function add:
+.Writing preferences at close time
+Under the *`def closing_cleanup__(self):`* function add:
```
I know "under" is the original wording, but wouldn't "in" or "inside" fit better?

smoe replied:
It is ambiguous. When I read it, I thought it was meant "below the function definition". From how you read it, I would suggest "directly under the line".

hansu commented on docs/src/gui/qtvcp-code-snippets.adoc:
```diff
@@ -69,30 +81,31 @@ self.SETTINGS_.setValue('myFloat', self.float_value)
-== Add a basic style editor
-
-Being able to edit a style on a running screen is convienant.
+== Add A Basic Style Editor
```
```suggestion
== Add a Basic Style Editor
```

And some further, I didn't mark all.

silopolis replied:
Could we leave it titles as they are for this PR and wait a later one once we decide on titles capitalization ?
Or can we decide now and I fix them all right away.

Again, personnal choice is to adopt "Sentence formatting" with only first char in capital, except other rightly capitalized words of course.

hansu replied:
Yes, just marked was catched my eye...

hansu commented on docs/src/gui/qtvcp-code-snippets.adoc:
```diff
@@ -160,75 +177,82 @@ def return_value(self, w, message):
-'SPEAK' is a key work, everything after it will be pronounced
+*`SPEAK`* is a keyword: _everything after it will be pronounced_.
 
 == ToolBar Functions.
```
```suggestion
== ToolBar Functions
```

hansu reviewed (commented):
:+1:

hansu commented:
The build error seems like a bug in asciidoc. So both anchors refer more or less to the same location.
It can be resolved by either this:
```diff
[[cha:qtvcp:vismach]]
= QtVismach

- [[sec:qtvcp:vismach:intro]]
== Introduction
```
or this:
```diff
[[cha:qtvcp:vismach]]
= QtVismach
+ Write something here
[[sec:qtvcp:vismach:intro]]
== Introduction
```

As the anchor `sec:qtvcp:vismach:intro` isn't used yet, I would tend to this option.

hansu commented:
When you have resolved the merge conflict - is it then ready to merge? Or do you have some squashing in mind?

silopolis commented:
Just rebased and pushed. Let's see if it builds then I'll do as much
squashing as possible (one commit per file ideal target), and push
again.

silopolis commented:
OK guys, squashed, rebased, and hopefully mergeable !

petterreinholdtsen commented:
[Jérémie Tarot]
> Most probably. This PR only really holds edits to QtVCP's adocs files and a
> handful of others. I never touched any man pages in any of my works, I
> believe.

If those are the only ones, I suspect making a new branch and doing 'git
cherry-pick' on these commits might be a simpler way forward:

c55c1fea5800b7d87cb5c9faaaa039d43481bc97
96d9526b51938dd8df22d3dd62cb3c241ca9e8b2
2a70df632cf523e9993aa1328220d16c44c4cf7e
78b2010860d11faa8b4bf9a4439e8afe26678879
4228d7aef330a6b9eb3e4e9342461a6211718f6f
681ed5116d54c8c367c69acbe6da6cff7ab6aeb2
c508e0fb8617997b4ea49c0334c9a6bf179082ce
a5303804d601ee57b2f789e3fd990d147ffb2e6f
10064207aa18c38b0ff6815898ef05559785a407

I suspect those are the ones you are interested in, as those are the
last ones in your docs/formating_fixes branch.

I tried rebasing on top of master, but that cause a lot of conflicts and
is most likely a very work intensive way forward.

-- 
Happy hacking
Petter Reinholdtsen

silopolis commented:
Le dim. 22 mai 2022 à 20:53, @smoe a écrit :

> @smoe <https://github.com/smoe> my question would rather be: why doesn't
> it finds the language where as it actually is in the
> source-highlight/local/ directory ?!
>
> So this builds for you, you are saying? Since the change is trivial and
> purely technical, this can be done easily for all languages. There is no
> point in delaying the merge over this, so I would just go and get to
> something that works and do anything more advanced later.
>
> No, I have to confess that I didn't try to build locally :/ Will do
tomorrow...

Again I never thought this would be merged before docs-devel reaches master
and still don't get why adding such a language sync breaking  change now ?!

[Forgot to send the message yesterday]

So, this neither builds locally.

Answer to my question, IIUC, could be that source-highlight is only used
for HTML rendering, whereas TeX `listings` package is used for PDF

A short dig into LaTeX code typesetting revealed that `listings` package
languages support may not be the best and, mainly, doesn't support CSS !

My proposal would be to switch to Pygments as a code highlighter for both
HTML and PDF docs.
AsciiDoc already supports it directly via the source-highlighter option in
asciidoc.conf.
LateX supports it via the `minted` package (https://www.ctan.org/pkg/minted),
as a replacement of `listings`.

Pygments support INI syntax out of the box, and forum member @blu-base has
already made (
https://forum.linuxcnc.org/20-g-code/37742-gcode-highlighting-with-pygments)
quite a nice G-code lexer (https://github.com/blu-base/gcodelexer). Only
left is HAL.

Down at the bottom of this rabbit hole, I discovered that Highlight (
http://www.andre-simon.de/doku/highlight/en/highlight.php) could also be a
nice option, but potentially with an even more involved migration...

At the end of this day, main issue for me is to understand that AsciiDoc
toolchain and our build system to finally find where and how to switch
between option and make some tests...

In the meantime, I'll switch CSS code blocks to HTML as suggested by @smoe
:'(

silopolis commented:
Le jeu. 15 sept. 2022 à 14:10, Hans Unzner ***@***.***> a
écrit :

> ***@***.**** commented on this pull request.
>
> I'll check that and commit it separately.
>

Awesome, thank you :)

silopolis commented:
Le mar. 20 sept. 2022 à 11:31, Hans Unzner ***@***.***> a
écrit :

> When you have resolved the merge conflict - is it then ready to merge? Or
> do you have some squashing in mind?
>

Planning to squash all review commits.
I'm on my style guide branch ATM. Tried to use worktrees again yesterday
evening and either I still doesn't understand some things, or Codium is not
ready for these. Rebasing a branch in a worktree has created a git mayhem
in editor so I'm back to the good old stash/co/co/pop Fu :/

Will push later today if squash plan is ok for you.
